test/CadlRanchProjects.Tests/type-model-inheritance-nested-discriminator.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using _Type.Model.Inheritance.NestedDiscriminator;
+using _Type.Model.Inheritance.NestedDiscriminator.Models;
+using AutoRest.TestServer.Tests.Infrastructure;
+using NUnit.Framework;
+
+namespace CadlRanchProjects.Tests
+{
+    public class type_model_inheritance_nested_discriminator : CadlRanchTestBase
+    {
+        [Test]
+        public Task Type_model_inheritance_nested_discriminator_getMissingDiscriminator() => Test(async (host) =>
+        {
+            var response = await new NestedDiscriminatorClient(host, null).GetMissingDiscriminatorAsync();
+            var value = response.Value;
+            Assert.AreEqual(1, value.Age);
+        });
+
+        [Test]
+        public Task Type_model_inheritance_nested_discriminator_putModel() => Test(async (host) =>
+        {
+            var value = new GoblinShark(1)
+            {
+                Kind = "shark",
+            };
```

Comment:
well it is cheating that we assign the internal properties in a test case because users of our generated libraries could never assign them.
Since the current implemented code is not correct, therefore without assigning this property will never let the test pass.
But still I suggest we change the implementation of our test cases to the correct way and let the test fail and wait for the real fix on the generator.
Therefore it should be
```suggestion
            var value = new GoblinShark(1);
```
here, and the test now will fail, just let it fail. It should pass when the generator fixes its issue.